src/editor/gridstack/gridstackLayout.ts

```diff
@@ -0,0 +1,196 @@
+import { Layout, Widget } from '@lumino/widgets';
+
+import { IIterator, ArrayIterator } from '@lumino/algorithm';
+
+import { Signal } from '@lumino/signaling';
+
+import {
+  GridStack,
+  GridHTMLElement,
+  GridStackNode,
+  GridItemHTMLElement
+} from 'gridstack';
+
+import 'gridstack/dist/gridstack.css';
+
+import { GridStackItem } from './gridstackItemWidget';
+
+import { DashboardView, DashboardCellView } from './../format';
+
+export class GridStackLayout extends Layout {
+  constructor(info: DashboardView) {
+    super();
+    this._margin = info.cellMargin;
+    this._cellHeight = info.cellHeight;
+    this._columns = info.numColumns;
+
+    this._gridItems = new Array<GridStackItem>();
+
+    this.gridItemChanged = new Signal<this, GridStackNode[]>(this);
+  }
+
+  readonly gridItemChanged: Signal<this, GridStackNode[]>;
```

Comment:
Usually it's a good practice to expose a `ISignal` instead of `Signal` as a getter.

And keep the private attribute as `Signal`.